File: android/app/src/main/java/com/smartexpense/android/domain/repository/NotificationRepository.java
package com.smartexpense.android.domain.repository;

import com.smartexpense.android.data.remote.dto.response.NotificationResponseDto;

import java.util.List;

public interface NotificationRepository {
    void getNotifications(ResultCallback<List<NotificationResponseDto>> callback);
    void markAsRead(String id, ResultCallback<Void> callback);
    void markAllAsRead(ResultCallback<Void> callback);
}

File: android/app/src/main/kotlin/com/smartexpense/android/domain/repository/NotificationRepository.kt
package com.smartexpense.android.domain.repository

import com.smartexpense.android.data.remote.dto.response.NotificationResponseDto

interface NotificationRepository {
    suspend fun getNotifications(): Result<List<NotificationResponseDto>>
    suspend fun markAsRead(id: String): Result<Unit>
    suspend fun markAllAsRead(): Result<Unit>
}
